src/modules/tunnel/unabto_tunnel_common.c

```diff
@@ -286,13 +286,13 @@ void unabto_tunnel_select_add_to_fd_set(fd_set* readFds, int* maxReadFd, fd_set*
 #if NABTO_ENABLE_TUNNEL_TCP
             if (tunnels[i].tunnelType == TUNNEL_TYPE_TCP){
                 if (tunnels[i].state == TS_FORWARD && tunnels[i].extReadState == FS_READ) {
-                    FD_SET(tunnels[i].tunnel_type_vars.tcp.sock, readFds);
-                    *maxReadFd = MAX(*maxReadFd, (int)(tunnels[i].tunnel_type_vars.tcp.sock));
+                    FD_SET(tunnels[i].tunnel_type_vars.tcp.sock.socket, readFds);
+                    *maxReadFd = MAX(*maxReadFd, (int)(tunnels[i].tunnel_type_vars.tcp.sock.socket));
```

Comment:
again we should make some functions which work on the struct unabto_tcp_socket type such that we do not rely on the struct having a socket variable which is an int.